Fix I2C NG includes

tore-espressif opened the pull request:
# ESP-BSP Pull Request checklist

> Note: For new BSPs create a PR with this [link](https://github.com/espressif/esp-bsp/compare/main...my-branch?quick_pull=1&template=pr_template_bsp.md).
- [x] Version of modified component bumped
- [x] CI passing

# Change description
* For BSP that exposed public function `i2c_master_bus_handle_t bsp_i2c_get_handle(void);` (korvo-2 and s3-eye), the `esp_driver_i2c` component must be listed in public dependencies
* On CoreS3 the camera is now enabled during `bsp_i2c_init()` call. Previously it was enabled during `bsp_display_new()`, but this caused problems if the users wanted to initialize the camera before LCD initialization

espzav commented on bsp/m5stack_core_s3/m5stack_core_s3.c:
```diff
@@ -88,6 +89,13 @@ esp_err_t bsp_i2c_init(void)
+     * All other peripherals (touch, audio, SD card...) are enabled in their init call (eg. bsp_sdcard_mount()),
+     * but we do not have a BSP init call for camera.
+     */
+    BSP_ERROR_CHECK_RETURN_ERR(bsp_enable_feature(BSP_FEATURE_CAMERA));
```
We can make it as public API like here: https://github.com/espressif/esp-bsp/blob/6c402325d9e2624608143ce39ead9c65bb9a7b34/bsp/m5stack_core_2/include/bsp/m5stack_core_2.h#L357C11-L357C29 
And then enable it in example or application.

tore-espressif replied:
mm, but then I cannot write 1 example for multiple BSPs... we did not consider this before... Let's discuss tomorrow

espzav replied:
I understand. I am thinking, that there will be nice to add some define to each BSP. Something like this:
```
#define BSP_ESP_BOX_3
```

And then we can use `#ifdef BSP_ESP_BOX_3` for special API in examples. I have similar problem in benchmark example [here](https://github.com/espressif/esp-bsp/blob/244b10bb4bd598178d27e1f31a5e10b190eacf9d/examples/display_lvgl_benchmark/main/main.c#L21)

tore-espressif commented:
@espzav 
* I removed the commit that changed `bsp_enable_feature` for now, so the other changes are not delayed
* I removed include of `soc/usb_pins.h` as this file will be removed in IDF6.0
